tests/scripts/all.sh

```diff
@@ -795,7 +795,81 @@ pre_generate_files() {
+# 2. Make configurations changes that are common to driver and main libraries.
+#    WARNING: both libraries MUST have the same set of PSA_WANT_ALG symbols,
+#    as they determine the ABI between them, so all changes to these symbols
+#    go here.
+#    2a. Call helper_libtestdriver1_adjust_config.
+#    2b. Any other change you need to make.
+# 3. Build the driver library: libtestdriver1.a.
+#    Just call helper_libtestdriver1_make_drivers, passing "$loc_accel_list".
+#    You may need to enable more algorithms here, typically hash algorithms
+#    when accelerating some signature algoritms (ECDSA, RSAv2). This is done
+#    by passing a 2nd argument listing the extra algorithms.
+#    Example:
+#      loc_extra_list="ALG_SHA_224 ALG_SHA_256 ALG_SHA_384 ALG_SHA_512"
+#      helper_libtestdriver1_make_drivers "$loc_accel_list" "$loc_extra_list"
+# 4. Configure and build the main libraries using drivers.
+#    4a. (optional) Call scripts/config.py full if you want.
+#    4b. Call scripts/config.py set MBEDTLS_PSA_CRYPTO_CONFIG
+#    4c. Make any other config adjustments you want. Typically you want to
+#        disable the modules that are being accelerated. You may need to also
+#        disable modules that depend on them or options that are not supported
+#        with drivers.
+#        WARNING: do NOT change any PSA_WANT symbol at that point, that should
+#        only be done in step 2 above.
+#    4d. Call helper_libtestdriver1_make_main "$loc_accel_list".
+# 5. Run the tests you want.
+
+# Adjust the configuration - for both libtestdriver1 and main library,
+# as they should have the same PSA_WANT macros.
+helper_libtestdriver1_adjust_config() {
+    # Disable ALG_STREAM_CIPHER and ALG_ECB_NO_PADDING to avoid having
+    # partial support for cipher operations in the driver test library.
+    scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_STREAM_CIPHER
+    scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_ECB_NO_PADDING
+}
 
+# Build the drivers library libtestdriver1.a (with ASan).
+#
+# Parameters:
+# 1. a space-separated list of things to accelerate;
+# 2. optional: a space-separate list of things to also support.
+# Here "things" are PSA_WANT_ symbols but with PSA_WANT_ removed.
+helper_libtestdriver1_make_drivers() {
+    loc_accel_flags=$( echo "$1 ${2-}" | sed 's/[^ ]* */-DLIBTESTDRIVER1_MBEDTLS_PSA_ACCEL_&/g' )
+    make -C tests libtestdriver1.a CFLAGS=" $ASAN_CFLAGS $loc_accel_flags" LDFLAGS="$ASAN_CFLAGS"
+}
+
+# Build the main libraries, programs and tests,
+# linking to the drivers library (with ASan).
+#
+# Parameters:
+# 1. a space-separated list of things to accelerate;
+# *. remaining arguments if any are passed directly to make
+#    (examples: lib, -C tests test_suite_xxx, etc.)
+# Here "things" are PSA_WANT_ symbols but with PSA_WANT_ removed.
+helper_libtestdriver1_make_main() {
+    loc_accel_list=$1
+    shift
```

Comment:
Do we need this operation here? 

> The number of arguments is reduced each time shift is executed and eventually becomes zero.

It seems we are skipping first argument. Why?